File: trunk/Source/Runtime/CPlusplus/Asset/Bibim/GameAsset.cpp
#include <Bibim/PCH.h>
#include <Bibim/GameAsset.h>

namespace Bibim
{
    GameAsset::GameAsset()
        : status(EmptyStatus),
          revision(0)
    {
    }

    GameAsset::~GameAsset()
    {
    }
}

File: tools/Bibim.Toolkit/Asset/GameAsset.cs
using System;
using System.Collections.Generic;
using System.Runtime.Serialization;
using System.Text;

namespace Bibim.Asset
{
    public abstract class GameAsset
    {
        #region Fields
        public static readonly string TextFileExtension = "asset";
        public static readonly string BinaryFileExtension = "ab";
        #endregion

        #region Properties
        public int ClassID
        {
            get { return ClassIDAttribute.GetClassID(GetType()); }
        }
        
        public GameAssetStatus Status
        {
            get;
            protected set;
        }

        public uint Revision
        {
            get;
            protected set;
        }

        public object Tag
        {
            get;
            set;
        }
        #endregion

        #region Constructor
        protected GameAsset()
        {
            Status = GameAssetStatus.Empty;
            Revision = 0;
        }
        #endregion
    }
}
